MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -32,340 +30,296 @@ namespace MediaBrowser.MediaEncoding.Encoder
-
-            path = newPaths.Item1;
-
-            if (!ValidateVersion(path, true))
-            {
-                throw new ResourceNotFoundException("ffmpeg version 3.0 or greater is required.");
-            }
-
-            var config = GetEncodingOptions();
-            config.EncoderAppPath = path;
-            ConfigurationManager.SaveConfiguration("encoding", config);
-
-            Init();
-        }
 
-        private bool ValidateVersion(string path, bool logOutput)
-        {
-            return new EncoderValidator(_logger, _processFactory).ValidateVersion(path, logOutput);
+            return rc;
         }
 
-        private void ConfigureEncoderPaths()
+        private string GetEncoderPath(string path)
         {
-            if (_hasExternalEncoder)
-            {
-                return;
-            }
-
-            var appPath = GetEncodingOptions().EncoderAppPath;
-
-            if (string.IsNullOrWhiteSpace(appPath))
-            {
-                appPath = Path.Combine(ConfigurationManager.ApplicationPaths.ProgramDataPath, "ffmpeg");
-            }
-
-            var newPaths = GetEncoderPaths(appPath);
-            if (string.IsNullOrWhiteSpace(newPaths.Item1) || string.IsNullOrWhiteSpace(newPaths.Item2) || IsSystemInstalledPath(appPath))
+            if (Directory.Exists(path))
             {
-                newPaths = TestForInstalledVersions();
+                return GetEncoderPathFromDirectory(path, "ffmpeg");
             }
 
-            if (!string.IsNullOrWhiteSpace(newPaths.Item1) && !string.IsNullOrWhiteSpace(newPaths.Item2))
+            if (File.Exists(path))
             {
-                FFMpegPath = newPaths.Item1;
-                FFProbePath = newPaths.Item2;
+                return path;
             }
 
-            LogPaths();
+            return null;
         }
 
-        private Tuple<string, string> GetEncoderPaths(string configuredPath)
+        private string GetEncoderPathFromDirectory(string path, string filename)
         {
-            var appPath = configuredPath;
-
-            if (!string.IsNullOrWhiteSpace(appPath))
+            try
             {
-                if (Directory.Exists(appPath))
-                {
-                    return GetPathsFromDirectory(appPath);
-                }
+                var files = FileSystem.GetFilePaths(path);
 
-                if (File.Exists(appPath))
-                {
-                    return new Tuple<string, string>(appPath, GetProbePathFromEncoderPath(appPath));
-                }
-            }
+                var excludeExtensions = new[] { ".c" };
 
-            return new Tuple<string, string>(null, null);
+                return files.FirstOrDefault(i => string.Equals(Path.GetFileNameWithoutExtension(i), filename, StringComparison.OrdinalIgnoreCase)
+                                                    && !excludeExtensions.Contains(Path.GetExtension(i) ?? string.Empty));
+            }
+            catch (Exception)
```

Comment:
Do checks instead of silently failing.

Reply:
Think this is ok as not interested in 'why' it may fail.  Just return null under exception.